lambda_src/drivers/destination_s3.py

```diff
@@ -31,7 +32,13 @@ def parse_destination_uri(destination: Text) -> Tuple[Text, Text]:
     LOG.debug(f'destination from header is {destination}')
     parsed_url = urlparse(destination)
     bucket = parsed_url.netloc
-    prefix = parsed_url.path.split('/', 2)[1] if parsed_url.path.count('/') >= 2 else ''
+        
+    parts = parsed_url.path.split('/')
+    prefix_parts = [part for part in parts if not part.startswith('%') and part]
+    strftime_parts = [time.strftime(part) for part in parts if part.startswith('%')]
+
+    prefix_with_time = "/".join(prefix_parts + strftime_parts)
```

Comment:
@sfc-gh-nlele There is a piece from our discussion misinterpreted here:

There are 3 parts to the prefix from the user 's3://bucket-name/<optional resource path>/<optional-date-format-string>' here:
1. Bucket name
2. The optional prefix path may or may not exist (if missing the data should go to the `data/ <optional-date-format-string>` path rather than directly in the bucket, if present then data goes into for e.g.: `bucket-name/a/b/c/%y/%m/%d/` )
3. The optional time format string also may or may not exist. If present data goes into e.g. `bucket-name/a/b/c/%y/%m/%d/`, if not then it goes into the prefix path, e.g. `s3://bucket-name/a/b/c`
4. If both the optional resource prefix path and date format string are missing then we just place the data in the `s3://bucket-name/data/` path.

@sfc-gh-afedorov Does this make sense?

Reply:
Yes, I see that from the code, but all these rules seem to add avoidable extrinsic complexity and most importantly won't support the thing that the user likely wants, which is what AWS does, data stored in e.g. —

~~~
/AWSLogs/123456789012/CloudTrail/ap-northeast-1/2020/03/02/123456789012_CloudTrail_ap-northeast-1_20200302T0020Z_Fo891lLUhaxnyLhj1.json.gz
~~~

in order to support this, we could allow the user to specify a prefix using `strftime` flags and parameters specifying account id (`{1}`) and region (`{2}`) like —

~~~
/AWSLogs/{1}/CloudTrail/{2}/%Y/%m/%d/{1}_CloudTrail_{2}_%Y%m%dT%H%MZ_
~~~

Not sure if we already support the parameters, but I'm OK with adding `strftime` first and then the parameters in a separate PR, if preferred.

Reply:
All Bruk wanted to the year, month, date options in the path and that was it. Additionally he said it would be good to have it in the file name but he was okay with not having it also.

Reply:
k, we can skip the parametrization for now if you'd like, but let's let the user put the strftime flags wherever they want...

Reply:
Where they want meaning before or after an optional prefix? So what you're saying is it's okay to have something like this?

```
'prefix' = 's3://my-bucket/a/%y/b/%d/%m/c'

-- or

'prefix' = 's3://my-bucket/a/%Y/b/%d/c/%m'
```

This would make the logic to collate the sfrftime variables and passing it to the strftime function unnecessarily complicated.

Reply:
it wudn't be to complicated if we apply it to the whole string I guess. @sfc-gh-nlele Could you please test strftime'ing the full path and see if that allows us to keep our logic simple?

Reply:
actually just tested and it seems to work just fine.
<img width="619" alt="image" src="https://user-images.githubusercontent.com/72515998/186711359-168b8779-13f2-42a3-95bd-f83b9ccb9c8a.png">

Reply:
yeah, I think `strftime` is kind of like `sprintf` [1] in that it is intended to print the whole string to another string but with certain flags replaced. I'm not 100% sure if there are times when someone might not want this replacement, but it seems easy enough to make it optional via something like a `destination-uri-raw` option, if it becomes necessary for some reason...

1. https://www.geeksforgeeks.org/sprintf-in-c/